### ISSUE TITLE
lost gradient during AdaptiveAvgPool1d backward

### ISSUE THREAD

**Issue by ailzhang** [pytorch/xla#912]
```
import torch
import torch_xla
import torch.nn as nn
import pdb

device = 'xla:0'
# device='cpu'
m = nn.AdaptiveAvgPool1d(1).to(device)
input = torch.rand(1, 3, 5, requires_grad=True, device=device)
output = m(input)
xla_gradOutput = output.clone().detach()
xla_gradOutput.requires_grad = True
xla_gradInputs = torch.autograd.grad(output, (input, ) + tuple(m.parameters()), xla_gradOutput, create_graph=True)
xla_gg = torch.autograd.grad(output.sum() + sum(map(lambda x: x.sum(), xla_gradInputs)),
        (input, xla_gradOutput),
        retain_graph=True)
pdb.set_trace()
print(xla_gg)
```

**Comment by dlibenzi** [comment]
Hopefully the gradient loss issues will be fixed with the `as_strided()` as view PR.

**Comment by ailzhang** [comment]
Confirmed this is fixed on master. Very likely it was fixed with `as_strided()` PR. :D Closing
